src/util/Options.js

```diff
@@ -132,6 +134,7 @@ class Options extends null {
    */
   static createDefault() {
     return {
+      closeTimeout: 5_000,
```

Comment:
Isn't this too long? Could it be shortened? Or is 5s minimum to ensure a complete close in most cases?

Reply:
Well, I have been using 2500ms as the max amount of time the WS should receive the close frame. Again the time taken might differ with the internet speed of the end-user. We can make 5s to 2.5s since I found 2500ms good enough for the close frame to be received. We can leave it to the users to go below or above that.

Reply:
That's alright then, I mean it doesn't matter anyways since its a "in case we never get a closing frame" timeout rather than a required one.

Reply:
> That's alright then, I mean it doesn't matter anyways since its a "in case we never get a closing frame" timeout rather than a required one.

Yes, and it's better at 5_000ms since giving it lesser time will make it emit close with 1011 when it was going to receive the close frame.
It would be a blunder if the close code which we were going to receive was `4004` but since the close timeout was so small it'll emit a close with 1011 instead of 4004 which can cause reconnect loops. It's not necessary that it'll wait 5000ms. It can receive the close frame in a few ms, so it's not that long.